experimental/PropertyDDS/packages/property-shared-tree-interop/src/schemaConverter.ts

```diff
@@ -0,0 +1,362 @@
+						extractContexts(property.properties);
+					}
+					if (property.context && property.context !== "single") {
+						referencedTypeIDs.add(
+							`${property.context}<${property.typeid}>` as TreeSchemaIdentifier,
+						);
+					}
+					if (TypeIdHelper.isPrimitiveType(property.typeid)) {
+						referencedTypeIDs.add(property.typeid);
+					}
+				}
+			}
+		};
+		extractContexts(schemaTemplate.properties);
+	}
+
+	for (const type of primitiveTypes) {
+		const typeid: TreeSchemaIdentifier = brand(type);
+		if (!referencedTypeIDs.has(typeid)) {
+			referencedTypeIDs.add(typeid);
+		}
+	}
+
+	// Now we create the actual schemas, since we are now able to reference the dependent types
+	for (const referencedTypeId of referencedTypeIDs.values()) {
+		if (lookupTreeSchema(repository, referencedTypeId) !== neverTree) {
+			continue;
+		}
+
+		const splitTypeId = TypeIdHelper.extractContext(referencedTypeId);
+		let typeSchema: NamedTreeSchema | undefined;
+
+		if (splitTypeId.context === "single") {
+			if (TypeIdHelper.isPrimitiveType(splitTypeId.typeid)) {
+				// @TODO for simplicity we convert it to native string
+				// if (splitTypeId.typeid === "String") {
+				//     // String is a special case, we actually have to represent it as a sequence
+				//     typeSchema = {
+				//             name: referencedTypeId,
+				//             localFields: new Map<LocalFieldKey, FieldSchema>([
+				//                 // TODO: What should be the key we use for the entries? Should this be standardized?
+				//                 ["entries" as LocalFieldKey, {
+				//                     kind: FieldKinds.sequence.identifier,
+				//                     types: new Set([
+				//                         // TODO: Which type do we use for characters?
+				//                     ]),
+				//                 }],
+				//             ]),
+				//             globalFields: new Set(),
+				//             extraLocalFields: emptyField,
+				//             extraGlobalFields: false,
+				//             value: ValueSchema.Nothing,
+				//         };
+				// } else {
+				let valueType: ValueSchema;
+				if (splitTypeId.isEnum) {
+					valueType = ValueSchema.Number;
+				} else if (
+					splitTypeId.typeid === "String" ||
+					splitTypeId.typeid.startsWith("Reference<") ||
+					splitTypeId.typeid === "Reference"
+				) {
+					valueType = ValueSchema.String;
+				} else if (booleanTypes.has(splitTypeId.typeid)) {
+					valueType = ValueSchema.Boolean;
+				} else if (numberTypes.has(splitTypeId.typeid)) {
+					valueType = ValueSchema.Number;
+				} else {
+					throw new Error(`Unknown primitive typeid: ${splitTypeId.typeid}`);
+				}
+
+				typeSchema = namedTreeSchema({
+					name: referencedTypeId,
+					extraLocalFields: emptyField,
+					value: valueType,
+				});
+				// }
+			} else {
+				if (splitTypeId.typeid === "NodeProperty") {
+					typeSchema = namedTreeSchema({
+						name: referencedTypeId,
+						extraLocalFields: fieldSchema(FieldKinds.optional),
```

Comment:
Should we be more restricitive here and only allow types which have been registerd via the PropertyFactory / are base types? At the moment, a user could put a type that has been independently registered into a NodeProperty.